src/components/AddOrEditSkillModale.tsx

```diff
@@ -16,12 +16,17 @@ const AddOrEditSkillModale = ({
 }: AddOrEditSkillModaleProps) => {
   const { t } = useContext(i18nContext);
   const [navState, setNavState] = useState("knowledge");
+  const [autoValidation, setautoValidation] = useState(false);
   const [skillLevel, setSkillLevel] = useState(skill?.level || 0);
   const [desireLevel, setDesireLevel] = useState(skill?.desire || 0);
 
   const onAddButtonClick = () => {
     callback({ ...skill, level: skillLevel, desire: desireLevel });
   };
+  useEffect(() => {
+    if (autoValidation)
+      onAddButtonClick();
+  }, [!autoValidation]);
```

Comment:
I think there is a simpler way to do this than adding a new variable. Since clicking on one of the buttons changes the desire level you could just do 
```ts
useEffect(() => {
    if (skillLevel != 0 && desireLevel != 0) {
      onAddButtonClick();
    }
  }, [desireLevel]);
```